tests/python/test_matrix.py

```diff
@@ -248,6 +245,20 @@ def func4(k: ti.i32):
     func4(10)
 
 
+@test_utils.test(require=ti.extension.dynamic_index,
+                 dynamic_index=True,
+                 debug=True)
+def test_matrix_non_constant_index():
+    _test_matrix_non_constant_index()
+
+
+@test_utils.test(require=ti.extension.dynamic_index,
+                 real_matrix=True,
```

Comment:
Turn on `dynamic_index=True` here?

Reply:
Actually we don't need to turn on that, because this code path doesn't rely on that option. That option is mainly to control alloca of local matrix before we have real matrix.